devices/temp_nix.go

```diff
@@ -28,6 +33,38 @@ func getTemps(temps map[string]int) map[string]error {
 			temps[label] = int(sensor.Temperature)
 		}
 	}
+
+	block, err := ghw.Block()
+	if err != nil {
+		log.Print("error getting block device info")
+		return nil
+	}
+
+	var sata = regexp.MustCompile(`sd.?`)
+	var nvme = regexp.MustCompile(`nvme*`)
+	for _, disk := range block.Disks {
+		switch t := disk.Name; {
```

Comment:
Haven't found a better way to create a switch depending on the disk type (sda / nvme)...

Reply:
A better and more idiomatic way to do it is following code:

```go
for _, disk := range block.Disks {
        fmt.Println(disk.Name)
        dev, err := smart.Open("/dev/" + disk.Name)
        if err != nil {
                fmt.Println(err)
                continue
        }

        switch sm := dev.(type) {
        case *smart.SataDevice:
                _, _ = sm.Identify()
        case *smart.ScsiDevice:
                _, _ = sm.Capacity()
        case *smart.NVMeDevice:
                _, _ = sm.ReadSMART()
        }
}
```

Reply:
thanks for the suggestion @anatol it's definitely a more elegant way to do this.
refactored in 98c69b7dacf85b8b25be909b33c333491ae481ad